src/Model/AbstractModel.php

```diff
@@ -149,7 +151,7 @@ public function setWriteConcern(int $writeConcern): void
     /**
      * @throws BindingResolutionException
      */
-    public function bsonSerialize(): object|array
+    public function bsonSerialize(): array|stdClass|Document
```

Comment:
```suggestion
    public function bsonSerialize(): array
```
Analisando o `\Mongolid\Query\ModelMapper::map`, aparentemente ele só pode devolver um array